client/header/activity-panel/index.js

```diff
@@ -260,35 +261,68 @@ class ActivityPanel extends Component {
 }
 
 export default withSelect( select => {
-	const { getReportItems, getReportItemsError, isReportItemsRequesting } = select( 'wc-api' );
+	const {
+		getCurrentUserData,
+		getReportItems,
+		getReportItemsError,
+		isReportItemsRequesting,
+		getReviews,
+		getReviewsTotalCount,
+		getReviewsError,
+		isGetReviewsRequesting,
+	} = select( 'wc-api' );
+	const userData = getCurrentUserData();
 	const orderStatuses = wcSettings.wcAdminSettings.woocommerce_actionable_order_statuses || [
 		'processing',
 		'on-hold',
 	];
 
+	let unreadOrders = null;
 	if ( ! orderStatuses.length ) {
-		return { unreadOrders: false };
-	}
+		unreadOrders = false;
+	} else {
+		const ordersQuery = {
+			page: 1,
+			per_page: 0,
+			status_is: orderStatuses,
+		};
 
-	const ordersQuery = {
-		page: 1,
-		per_page: 0,
-		status_is: orderStatuses,
-	};
+		const totalOrders = getReportItems( 'orders', ordersQuery ).totalResults;
+		const isOrdersError = Boolean( getReportItemsError( 'orders', ordersQuery ) );
+		const isOrdersRequesting = isReportItemsRequesting( 'orders', ordersQuery );
 
-	const totalOrders = getReportItems( 'orders', ordersQuery ).totalResults;
-	const isError = Boolean( getReportItemsError( 'orders', ordersQuery ) );
-	const isRequesting = isReportItemsRequesting( 'orders', ordersQuery );
+		if ( ! isOrdersError && ! isOrdersRequesting ) {
+			if ( totalOrders > 0 ) {
+				unreadOrders = true;
+			} else {
+				unreadOrders = false;
+			}
+		}
+	}
 
-	let unreadOrders = null;
+	let numberOfReviews = null;
+	let unreadReviews = false;
+	if ( 'yes' === wcSettings.reviewsEnabled ) {
+		const reviewsQuery = {
+			order: 'desc',
+			orderby: 'date_gmt',
+			page: 1,
+			per_page: 1,
+		};
+		const reviews = getReviews( reviewsQuery );
+		const totalReviews = getReviewsTotalCount( reviewsQuery );
+		const isReviewsError = Boolean( getReviewsError( reviewsQuery ) );
+		const isReviewsRequesting = isGetReviewsRequesting( reviewsQuery );
 
-	if ( ! isError && ! isRequesting ) {
-		if ( totalOrders > 0 ) {
-			unreadOrders = true;
-		} else {
-			unreadOrders = false;
+		if ( ! isReviewsError && ! isReviewsRequesting ) {
+			numberOfReviews = totalReviews;
+			unreadReviews =
+				reviews.length &&
+				reviews[ 0 ] &&
```

Comment:
Maybe this second check should be `reviews[ 0 ]. date_created_gmt` because `new Date( undefined + 'Z')` is "Invalid Date".

Reply:
Fixed in: 11bc68aa3e5f18f2d54aa37fbabe56f54f02278c.